runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -4715,7 +4715,7 @@ inline void generateInlinedCheckCastOrInstanceOfForClass(TR::Node* node, TR_Opaq
    generateLoadJ9Class(node, j9class, j9class, cg);
 
    // Equality test
-   if (!fej9->isAbstractClass(clazz))
+   if (!fej9->isAbstractClass(clazz) || node->getOpCodeValue() == TR::icall/*TR_checkAssignable*/)
```

Comment:
Can you add a comment to explain why the equality test can be skipped for non-abstract classes but is required for TR_checkAssignable?  Having the explanation in the code would make it clear to future readers.